lib/tests/test.js

```diff
@@ -111,22 +112,35 @@ class Test {
   }
 
   onReady(callback) {
+    const self = this;
     if (this.ready) {
       return callback();
     }
+    if (this.error) {
+      return callback(this.error);
+    }
     this.events.once('ready', () => {
+      self.events.removeListener('deployError', () => {});
       callback();
     });
+    this.events.once('deployError', (err) => {
+      self.events.removeListener('ready', () => {});
```

Comment:
You need to remove the exact same callback btw. So create another function and pass it to the once and the remove.

Reply:
u're right, good catch